db/commitlog/commitlog.cc

```diff
@@ -973,11 +986,22 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
             overhead += descriptor_header_size;
         }
 
-        auto a = align_up(s + overhead, _alignment);
+        s += overhead;
+        // add bookkeep data reqs. 
+        s += (s / (_alignment - detail::sector_overhead_size)) * detail::sector_overhead_size;
```

Comment:
that's repeated logic, can it be captured by a helper?

Reply:
Sure.